imap_processing/run_processing.py

```diff
@@ -0,0 +1,122 @@
+"""Serves as a command line utility to run the processing for a specific
+instrument and data level.
+
+Use
+---
+    python run_processing.py <instrument> <data_level>
+"""
+
+import argparse
+
+from imap_processing import instruments, processing_levels
+
+
+def _parse_args():
+    """Parse the command line arguments.
+
+    Returns
+    -------
+    args : argparse.Namespace
+        An object containing the parsed arguments and their values
+    """
+    description = (
+        "This command line program invokes the processing pipeline "
+        "for a specific instrument and data level. Example usage: "
+        '"python run_processing swe l1a".'
+    )
+
+    instrument_help = (
+        "The instrument to process. Acceptable values are: " '"swe", and "codice".'
+    )
+    level_help = (
+        "The data level to process. Acceptable values are: "
+        '"l0", "l1a", "l1b", "l1c", and "l2".'
+    )
+
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("instrument", type=str, help=instrument_help)
+    parser.add_argument("level", type=str, help=level_help)
+
+    args = parser.parse_args()
+
+    return args
+
+
+def _validate_args(args):
+    """Ensure that the provided arguments are valid before kicking off the
+    processing.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        An object containing the parsed arguments and their values
+    """
+    assert (
+        args.instrument in instruments
+    ), f"{args.instrument} is not a supported instrument"
+    assert (
+        args.level in processing_levels[args.instrument]
+    ), f"{args.level} is not a supported data level"
+
+
+class ProcessInstrument:
```

Comment:
I like this idea. Do you want to make this an abstract base class for a single instrument that then other people would override?

```python
class ProcessInstrument(abc.abc):
    def process(level):
        raise NotImplementedError


Codice(ProcessInstrument):
    # Not sure if we want this defined here, or in instrument modules?
    def process(level):
        # instruments would override this method themselves.

# lookup table to dispatch to each of the methods?